test/functional/ex_cmds/cd_spec.lua

```diff
@@ -0,0 +1,54 @@
+-- Specs for :cd, :tcd, :lcd
+
+local helpers = require('test.functional.helpers')
+local nvim, execute, eq, clear, eval, feed =
+  helpers.nvim, helpers.execute, helpers.eq, helpers.clear, helpers.eval, helpers.feed
+
+describe(':cd :tcd', function()
+  before_each(clear)
+
+  it('sets to local directory of the current tab', function()
+    -- Store the initial working directory
+    local globalDir = eval('getcwd()')
+
+    -- Create a new tab first and verify that is has the same working dir
+    execute('tabnew')
+    assert.is_same(eval('getcwd()'), globalDir)
+
+    -- Change tab-local working directory and verify it is different
+    execute('tcd test')
```

Comment:
Missing `tchdir` test. This is a separate command, so you also need to verify that this spelling works.